session/session.go

```diff
@@ -2236,6 +2239,9 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 		metrics.ResourceGroupQueryTotalCounter.WithLabelValues(resourceGroupName).Inc()
 	}
 
+	// Restore the resource group for the session
+	s.GetSessionVars().ResourceGroupName = originalResourceGroup
```

Comment:
Better to make this a defer before `compiler.Compile` to ensure the session var can be reset back even if this function return early when meets error.

Reply:
Good suggestion.